mempool/clist_mempool.go

```diff
@@ -446,17 +446,22 @@ func (mem *CListMempool) TxsAvailable() <-chan struct{} {
 }
 
 func (mem *CListMempool) notifyTxsAvailable() {
-	if mem.Size() == 0 {
-		panic("notified txs available but mempool is empty!")
-	}
-	if mem.txsAvailable != nil && !mem.notifiedTxsAvailable {
-		// channel cap is 1, so this will send once
-		mem.notifiedTxsAvailable = true
-		select {
-		case mem.txsAvailable <- struct{}{}:
-		default:
+	go func() {
+		if mem.Size() == 0 {
+			mem.logger.Error("notified txs available but mempool is empty!")
+			return
 		}
-	}
+		if mem.txsAvailable != nil && !mem.notifiedTxsAvailable {
+			// channel cap is 1, so this will send once
+			mem.notifiedTxsAvailable = true
+			time.Sleep(mem.config.Consensus.TimeoutCommit)
```

Comment:
why `TimeoutCommit` though and not some configurable interval (say 1s)?

Reply:
if 1 sec, then we create a block each second. But in consensus engine commit time out is set to 5 (default). I believe commit timeout should be same regardless of create_empty_block flag is set or not.

Reply:
I am not talking about changing TimeoutCommit. But rather using another variable

Reply:
You mean adding new item to mempool config?

Reply:
Yeah

Reply:
`no_empty_blocks_wait`, Is it good name for this config? Any suggestion?

Reply:
Another way, What's your idea I pass `TimeoutCommit` an new param to the constructor? So I don't need to pass whole config here.